server/middlewares/resetpassword.js

```diff
@@ -0,0 +1,101 @@
+import notification from '../helpers/notifications';
+import Authenticate from '../helpers/auth';
+import model from '../models';
+
+const { User } = model;
+
+const { decode, encodeEmail } = Authenticate;
+
+const ResetPasswordMiddleware = {
+  async validateEmail(req, res, next) {
+    try {
+      const userEmail = await User.findOne({
+        where: {
+          email: req.body.email,
+        },
+      });
+
+      if (!userEmail) {
+        return res.status(404).json({
+          error: 'Email not found',
+        });
+      }
+      req.user = userEmail;
+      return next();
+    } catch (error) {
+      return res.status(500).json({
+        message: error.message,
+      });
+    }
+  },
+
+  createToken(req, res, next) {
+    const token = encodeEmail(req.body.email);
+    req.token = token;
+    return next();
+  },
+
+  async mailer(req, res) {
+    try {
+      const linkUrl = `${req.protocol}://${req.get('host')}/api/v1/auth/reset/${
+        req.token
+      }/password`;
+
+      await notification.forgetPassword(req.body.email, linkUrl);
+
+      return res.status(200).json({
+        status: 200,
+        data: [
+          {
+            message: 'Check your email for password reset link',
+            email: req.body.email,
+            token: req.token,
+          },
+        ],
+      });
+    } catch (error) {
+      return res.status(500).json({
+        message: error.message,
+      });
+    }
+  },
+
+  async validateToken(req, res, next) {
+    try {
+      req.data = decode(req.params.token);
+    } catch (error) {
+      if (error) {
+        return res.status(406).json({
+          status: 406,
```

Comment:
Why are we sending `406` here? @vic3king

Reply:
@senor I used that error to catch a situation for any reason where the token being decoded fails.